<!DOCTYPE html>
<html>
    <head>
        <title>PATRIC Genome Report | {{meta.genome_name}}</title>

        <link href="https://fonts.googleapis.com/css?family=Work+Sans:300,400,500,600,700,800,900" rel="stylesheet">

        <style>
            body {
                font-family: 'Work Sans', sans-serif;
                color: #444;
            }
            header { padding-bottom: .5in; }
            section { margin-bottom: .5in; }

            a {
                text-decoration: none;
                color: #0d78ef;
            }
            a:hover { text-decoration: underline; }
            h2 { font-size: 1.45em; }
            h2, h3 { font-weight: 500; }
            h2 small { color: #aaa; }
            .pull-left { float: left; }
            .pull-right { float: right; }

            table-num,
            table-ref,
            fig-num,
            fig-ref {
                font-weight: 600;
            }

            /* tables */
            table {
                margin: .1in 0 .5in 0;
                border-collapse: collapse;
                page-break-inside: avoid;
            }
            .table-header {
                color: #fff;
                background: #196E9C;
                padding: 5px;
                text-align: left;
            }
            .table-header th { font-weight: 400; }
            .row-header { border-bottom: 1px solid #196E9C; font-weight: 600; }
            th, td { padding: 5px; }
            th  { border-bottom: 2px solid #196E9C; }
            tr:last-child { border-bottom: 1px solid #196E9C; }

            .kv-table,
            .kv-table-2 {
                text-align: left;
            }

            .kv-table-2 td:nth-child(2) {
                border-right: 1px solid rgb(172, 172, 172)
            }

            .kv-table td:first-child,
            .kv-table-2 td:first-child,
            .kv-table-2 td:nth-child(3) {
                font-weight: 700;
            }

            table td.align-right {
                text-align: right;
            }

            .kv-table-2 td:nth-child(2) { padding-right: 10px; }
            .kv-table-2 td:nth-child(3) { padding-left: 10px; }
            .lg-table { width: 80%; }
            .med-table { width: 60%; }
            .sm-table { width: 40%; }
            .xs-table {width: 20%; }

            .center { margin-left: auto; margin-right: auto; }

            .logo {
                width: 2.5in;
                border-right: 3px solid #777;
                padding-right: 10px;
                margin-right: 10px;
            }
            .title {
                padding: 17px 0 0 0px;
                font-size: 1.3em;
                color: #777;
            }
            .report-info {
                text-align: right;
                margin-right: 30px;
                font-size:.8em;
                color: #777;
            }
            .report-info span {
                font-weight: 600;
            }

            .main-img {
                width: 250px;
                margin-right: .2in;
            }

            .subsystem-fig {
                width: 8in;
            }

            .circular-view-fig {
                width: 90%;
            }


            .tree-fig {
                width: 1100px;
                height: 150px;
            }

            .clearfix:after {
                visibility: hidden;
                display: block;
                font-size: 0;
                content: " ";
                clear: both;
                height: 0;
            }
            .clearfix { display: inline-block; }

            * html .clearfix { height: 1%; }
            .clearfix { display: block; }
        </style>

    </head>
    <body>
        <header>
            <!--
            <div class="pull-left">
                <img src="../patric-logo.svg" class="logo pull-left">
                <div class="title pull-left">Genome Report</div>
            </div>
            -->

            <div class="report-info pull-right">
                <span>Report Date:</span> {{reportDate}}<br>
                <span>Organism:</span> {{meta.genome_name}}
            </div>
        </header>

        <div class="clearfix"></div>

        {{#or wiki comments}}
        <section>
            <h2>Introduction</h2>

            <p>
                <img src="{{wiki.imageSource}}" class="pull-left main-img"/>
                <span class="pull-left">{{{wiki.text}}}</span>
            </p>

            <p>
                {{meta.comments}}
            </p>
        </section>
        {{/or}}

        <section>
            <h2>Genome Assembly</h2>
            {{#if gto.job_data.assembly.job_id}}
                <p>
                    {{meta.genome_name}} ran under job ID {{gto.job_data.assembly.job_id}} at PATRIC[1].
                    The assembly job started at {{datetime gto.job_data.assembly.start_time}} and completed at {{datetime gto.job_data.assembly.completion_time}}
                    after {{elapsed gto.job_data.assembly.elapsed_time}}.
                    The {{gto.job_data.assembly.parameters.recipe}} assembly strategy was selected.
                    The auto strategy runs three assembly strategies that include Velvet [3], IDBA [4] and Spades [5], each of which is given
                    an assembly score by ARAST, an in-house script. The minimum contig length was {{gto.job_data.assembly.parameters.min_contig_len}}bp
                    and smaller contigs were not included in the assembly. The minimum contig coverage was
                    {{gto.job_data.assembly.min_contig_cov}}
                    and contigs with less coverage were not included in the assembly.
                    Also add reference to QUAST. ARAST ranked the "{{gto.job_data.assembly.attributes.chosen_assembly}}"
                    assembly as the highest qualitye.
                </p>
            {{/if}}
            <p>
                The assembled genome has {{addCommas meta.contigs 0}} contigs,
                with the total length of ~{{basePairs meta.genome_length}}
                and %GC of {{default (toFixed meta.gc_content 2) 0}}% (<table-ref></table-ref>).
            </p>

            <table class="sm-table kv-table center">
                <thead class="table-header">
                    <tr>
                        <th colspan="4">
                            <table-num></table-num> Assembly Details
                        </th>
                    </tr>
                </thead>

                <tbody>
                    <tr>
                        <td>Contigs</td>
                        <td class="align-right">{{addCommas meta.contigs 0}}</td>
                    </tr>
                    <tr>
                        <td>GC Content</td>
                        <td class="align-right">{{default (toFixed meta.gc_content 2) 0}}</td>
                    </tr>
                    <tr>
                        <td>Plasmids</td>
                        <td class="align-right">{{addCommas meta.plasmids 0}}</td>
                    <tr>
                    </tr>
                        <td>Contig L50</td>
                        <td class="align-right">{{default meta.genome_metrics.L50 0}}</td>
                    </tr>
                    <tr>
                        <td>Genome Length</td>
                        <td class="align-right">
                            {{addCommas meta.genome_length 0}} Bp
                        </td>
                    </tr>
                    <tr>
                        <td>Contig N50</td>
                        <td class="align-right">
                            {{addCommas meta.genome_metrics.N50 0}}
                        </td>
                    </tr>
                    <tr>
                        <td>Chromosomes</td>
                        <td class="align-right">{{addCommas meta.chromosomes 0}}</td>
                    </tr>
                </tbody>
            </table>
        </section>



        <section>
            <h2>Genome Annotation</h2>
            <p>
                The Genome Annotation Service in PATRIC [1] uses the RAST tool kit (RASTtk) [6] to
                provide annotation of genomic features. The job for {{meta.genome_name}}
                ran under job number {{gto.job_data.annotation.job_id}}.
                The annotation job started at {{datetime gto.job_data.annotation.start_time}}
                and completed at {{datetime gto.job_data.annotation.completion_time}}
                after {{elapsed gto.job_data.annotation.elapsed_time}} seconds.
                This genome was described to be in the domain
                {{#eq gto.domain 'B'}}Bacteria{{/eq}}{{#eq gto.domain 'A'}}Archaea{{/eq}}
                with genetic code {{gto.genetic_code}}. It was assigned genome identifier
                {{gto.id}}. The taxonomy of {{meta.genome_name}} is:
            </p>

            <div style="width: 90%;" class="center">
                    {{#each gto.ncbi_lineage}} {{ this.[0] }} {{#unless @last }} > {{/unless }} {{/each}}
            </div>
            <br>


            <p>
                The annotations includes
                    {{addCommas (get 'count' 'CDS' annotationMeta) 0}} CDS,
                    {{addCommas (get 'count' 'tRNA' annotationMeta) 0}} tRNAs and
                    {{addCommas (get 'count' 'rRNA' annotationMeta) 0}} rRNAs.

                All annotated genome features for this genome are summarized in <table-ref></table-ref>.
            </p>

            <p>
                The functional annotation included
                    {{addCommas (get 'count' 'Hypothetical proteins' proteinFeatures) 0}} hypothetical proteins and
                    {{addCommas (get 'count' 'Proteins with functional assignments' proteinFeatures) 0}} proteins with functional assignments.
                Furthermore,
                    {{addCommas (get 'count' 'Proteins with EC number assignments' proteinFeatures) 0}} proteins were assigned EC numbers and
                    {{addCommas (get 'count' 'Proteins with Pathway assignments' proteinFeatures) 0}} proteins were mapped to KEGG pathways.

                A breakdown of the proteins that have been annotated for this isolate is provided in <table-ref></table-ref>.
            </p>

            <table class="sm-table kv-table center">
                <thead class="table-header">
                    <tr>
                        <th colspan="2"><table-num></table-num> Annotated Genome Features</th>
                    </tr>
                </thead>

                <tbody>
                    <tr class="row-header"><td></td> <td class="align-right">Count</td></tr>
                    {{#each annotationMeta}}
                        <tr>
                            <td>{{name}}</td>
                            <td class="align-right">
                                {{addCommas count 0}}
                            </td>
                        </tr>
                    {{/each}}
                </tbody>
            </table>


            <table class="med-table kv-table center">
                <thead class="table-header">
                    <tr>
                        <th colspan="2"><table-num></table-num> Protein Features</th>
                    </tr>
                </thead>

                <tbody>
                    <tr class="row-header"><td></td> <td class="align-right">Count</td></tr>
                    {{#each proteinFeatures}}
                    <tr>
                        <td>{{name}}</td>
                        <td class="align-right">
                            {{addCommas count 0}}
                        </td>
                    </tr>
                    {{/each}}
                </tbody>
            </table>

            <p>
                A graphical display of the distribution of the CDS on each strand, the repeat_regions,
                GC content and GC skew are provided (<fig-ref></fig-ref>).
            </p>

            <fig-num></fig-num>


            <div class="circular-view-fig center" >{{{circularViewSVG}}}</div>

        </section>


        <section>
            <h2>Genome Quality</h2>
            <p>
                Based on the analysis of genome assembly and annotation statistics and comparing
                them to other closely related genomes available in PATRIC for the same species, the
                overall genome seems to be of <i>{{meta.genome_quality}}</i> quality.
            </p>
            <p>
                {{#isnt meta.genome_quality 'Good'}}
                    The reasons affecting the quality of genome follows:
                {{/isnt}}
                <i>{{join meta.genome_quality_flags ', '}}</i>
            </p>


        </section>


        <section>
            <h2>Specialty Genes</h2>
            <p>
                PATRIC [1] annotation also includes identification of proteins that have homology to
                known virulence factors, drug targets, and antibiotic resistance genes (Reference
                the sources).  A breakdown of the distribution is provided (<table-ref></table-ref>).
            </p>

            <table class="sm-table kv-table center">
                <thead class="table-header">
                    <tr>
                        <th colspan="3"><table-num></table-num> Specialty Genes</th>
                    </tr>
                </thead>

                <tbody>
                    <tr class="row-header">
                        <td></td>
                        <td>Source</td>
                        <td class="align-right">Genes</td>
                    </tr>
                    {{#each specialtyGenes}}
                    <tr>
                        <td>{{type}}</td>
                        <td>{{source}}</td>
                        <td class="align-right">{{addCommas genes 0}}</td>
                    </tr>
                    {{/each}}
                </tbody>
            </table>
        </section>



        {{#amr}} <!-- only shown if data is available -->
        <section>
            <h2>Antimicrobial Resistance Analysis</h2>
            <table class="lg-table center">
                <thead class="table-header">
                    <tr>
                        <th colspan="2"><table-num></table-num> Antimicrobial Resistance Phenotypes</th>
                    </tr>
                </thead>

                <tbody>
                    <tr class="row-header">
                        <td></td>
                        <td>Antibiotics</td>
                    </tr>
                    <tr>
                        <td>Resistant</td>
                        <td>{{join resistant ', '}}</td>
                    </tr>
                    <tr>
                        <td>Susceptible</td>
                        <td>{{join susceptible ', '}}</td>
                    </tr>
                </tbody>

            </table>

            <h2>Antimicrobial Resistance Genes</h2>

            [Incomplete]
        </section>
        {{/amr}}



        <section>
            <h2>Subsystem Analysis</h2>
            <p>
                A subsystem is a set of protein functions that are related. Frequently, subsystems
                represent the collection of functional roles that make up a metabolic pathway, a complex
                (e.g., the ribosome), or a class of proteins (e.g., two-component signal-transduction
                proteins or AMR genes) [Overbeek].  An overview of the subsystems for this genome
                is provided in <fig-ref></fig-ref>.
            </p>

            <fig-num></fig-num>
            <div class="center subsystem-fig" >{{{subsystemSVG}}}</div>
        </section>


        <section>
            <h2>Phylogenetic Analysis</h2>

            <p>
                PATRIC [1] provides a phylogentic tree building service.  The makes use of
                several third-party tools. These include, BLAST[1], MCL[2], Muscle[3],
                hmmbuild[4], hmmsearch[4], Gblocks[5], FastTree[6], and RAxML[7].
                The pipeline begins with thegi set of ingroup genome protein files.
                These are filtered to remove duplicate species, resulting in a distinct-species
                subset of the ingroup genomes. This is done to reduce biasing the homolog
                sets with overrepresented species. BLAST searches are used to find
                bi-directional best hit protein pairs between genomes, and these bidirectional
                best hit pairs are clustered using MCL. Clusters containing members from at least
                half of the distinct genomes are chosen as initial, or seed, homolog sets. These
                seed sets are expanded to include members from all ingroup and outgroup taxa using
                tools from the HMMer suite. A hidden Markov Model (HMM) is built from each seed
                set using hmmbuild. These HMMs are used to search each genome, with hmmsearch,
                to find the best match from each genome for each homolog set model. The final,
                expanded, homolog sets are created from the hmmsearch results. Homolog sets representing
                fewer than 80% of ingroup genomes are removed. The remaining sets are aligned
                using Muscle, and the alignments are trimmed using GBlocks.
                The trimmed alignments are concatenated and this concatenated alignment is
                used to build the main tree with either RAxML or FastTree.
            </p>

            <p>
            [Tree will go here]
            </p>
            <!--
                <h3>Comparison to Reference Genomes in the Same Genus</h3>
                <fig-num></fig-num>
                <img src="../{{meta.genome_id}}/{{meta.genome_id}}-tree.svg" class="tree-fig" />
            -->
        </section>



        {{#if proteinFamily}} <!-- only shown if data is available -->
        <section>
            <h2>Unique Protein Families</h2>

            <p>
                In <table-ref></table-ref> we present protein families that the isolate has which
                all reference genomes lack.
            </p>


            <table class="lg-table center">
                <thead class="table-header">
                    <tr>
                        <th colspan="3"><table-num></table-num> Core Protein Families Missing in the Genome</th>
                    </tr>
                </thead>

                <tbody>
                    <tr class="row-header">
                        <td>Protein Family ID</td>
                        <td>Description</td>
                    </tr>
                    {{#proteinFamily}}
                    <tr>
                        <td>{{family_id}}</td>
                        <td>{{family_product}}</td>
                    </tr>
                    {{/proteinFamily}}
                </tbody>
            </table>
        </section>
        {{/if}}

    </body>
</html>