<!DOCTYPE html>
<html>
    <head>
        <title>PATRIC Genome Report | {{meta.genome_name}}</title>

        <link href="https://fonts.googleapis.com/css?family=Work+Sans:300,400,500,600,700,800,900" rel="stylesheet">

        <style>
            body {
                font-family: 'Work Sans', sans-serif;
                color: #444;
            }
            header { padding-bottom: .5in; }
            section { margin-bottom: .5in; }

            a {
                text-decoration: none;
                color: #0d78ef;
            }
            a:hover { text-decoration: underline; }
            h2 { font-size: 1.45em; }
            h2, h3 { font-weight: 500; }
            h2 small { color: #888; font-size: 0.65em; }
            .pull-left { float: left; }
            .pull-right { float: right; }
            sup { display: inline-block; }

            table-num,
            table-ref,
            fig-num,
            fig-ref {
                font-weight: 600;
            }

            /* tables */
            table {
                margin: .1in 0 .5in 0;
                border-collapse: collapse;
                page-break-inside: avoid;
            }
            .table-header {
                color: #fff;
                background: #196E9C;
                padding: 5px;
                text-align: left;
            }
            .table-header th { font-weight: 400; }
            .row-header { border-bottom: 1px solid #196E9C; font-weight: 600; }
            th, td { padding: 5px; }
            th  { border-bottom: 2px solid #196E9C; }
            tr:last-child { border-bottom: 1px solid #196E9C; }

            .kv-table,
            .kv-table-2 {
                text-align: left;
            }

            .kv-table-2 td:nth-child(2) {
                border-right: 1px solid rgb(172, 172, 172)
            }

            .kv-table td:first-child,
            .kv-table-2 td:first-child,
            .kv-table-2 td:nth-child(3) {
                font-weight: 700;
            }

            table td.align-right {
                text-align: right;
            }

            .kv-table-2 td:nth-child(2) { padding-right: 10px; }
            .kv-table-2 td:nth-child(3) { padding-left: 10px; }
            .lg-table { width: 80%; }
            .med-table { width: 60%; }
            .sm-table { width: 40%; }
            .xs-table {width: 20%; }

            .center { margin-left: auto; margin-right: auto; }

            .logo {
                width: 2.5in;
                border-right: 3px solid #777;
                padding-right: 10px;
                margin-right: 10px;
            }
            .title {
                padding: 17px 0 0 0px;
                font-size: 1.3em;
                color: #777;
            }
            .report-info {
                text-align: right;
                margin-right: 30px;
                font-size:.8em;
                color: #777;
            }
            .report-info span {
                font-weight: 600;
            }

            .main-img {
                width: 250px;
                margin-right: .2in;
            }

            .subsystem-fig {
                width: 8in;
            }

            .circular-view-fig {
                width: 80%;
            }

            .tree-fig {
                width: 70%;
            }

            ol.references li {
                margin-bottom: 1em;
            }

            .clearfix:after {
                visibility: hidden;
                display: block;
                font-size: 0;
                content: " ";
                clear: both;
                height: 0;
            }
            .clearfix { display: inline-block; }

            * html .clearfix { height: 1%; }
            .clearfix { display: block; }
        </style>

    </head>
    <body>
        <header>
            <!--
            <div class="pull-left">
                <img src="../patric-logo.svg" class="logo pull-left">
                <div class="title pull-left">Genome Report</div>
            </div>
            -->

            <div class="report-info pull-right">
                <span>Report Date:</span> {{currentDate}}<br>
                <span>Organism:</span> {{meta.genome_name}}
            </div>
        </header>
        <div class="clearfix"></div>


        <!--
          ## Summary
        -->
        <section>
            <h2>Summary</h2>
            {{#if gto.job_data.assembly.job_id}}
                Reads for {{meta.genome_name}} were submitted to the comprehensive genome analysis
                service at PATRIC<ref>Wattam, et al. 2017</ref>.
            {{else}}
                An assembled genome for {{meta.genome_name}} was submitted to the comprehensive genome analysis
                service at PATRIC<ref>Wattam, et al. 2017</ref>.
            {{/if}}
            Based on the annotation statistics and a comparison to other genomes in PATRIC within
            this same species, this genome appears to be of {{meta.genome_quality}} quality.
            {{#isnt meta.genome_quality 'Good'}}
                The reasons affecting the quality of genome follows:
                <i>{{join meta.genome_quality_flags ', '}}</i>.
            {{/isnt}}

            Details of the analysis, including genes of
            interest (Specialty Genes){{#meta.subsystem_summary}}, a functional categorization (Subsystems), {{/meta.subsystem_summary}}
            and a phylogenetic tree (Phylogenetic Analysis) are provided below.
        </section>



        <!--
          ## Genome Assembly
        -->
        <section>
            <h2>Genome Assembly</h2>

            {{#if gto.job_data.assembly.job_id}}

                <p>
                    {{meta.genome_name}} was assembled using
                    {{prettyRecipe gto.job_data.assembly.attributes.chosen_assembly}}<ref class="assembly-method"></ref>. There were
                    {{commas meta.contigs 0}} contig{{plural meta.contigs}}, an estimated genome length of {{commas meta.genome_length 0}} bp,
                    and an average G+C content of {{default (toFixed meta.gc_content 2) 0}}%.

                    The N50 length, which is defined as the shortest sequence length at 50% of
                    the genome, is {{commas meta.genome_metrics.N50 0}} bp.   The L50 count, which is defined as
                    the smallest number of contigs whose length sum produces N50, is {{commas meta.genome_metrics.L50 0}} (<table-ref></table-ref>).
                </p>

            {{else}}
                <p>
                    An assembled genome was submitted to the Comprehensive Genome Analysis service.
                    This assembled genome had
                        {{commas meta.contigs 0}} contig{{plural meta.contigs}},
                    with the total length of
                        {{commas meta.genome_length 0}} bp
                    and an average G+C content of
                        {{default (toFixed meta.gc_content 2) 0}}% (<table-ref></table-ref>).
                </p>

            {{/if}}


            <table class="{{#if gto.job_data.assembly.job_id}}med-table{{else}}sm-table{{/if}} kv-table center">
                <thead class="table-header">
                    <tr>
                        <th colspan="4">
                            <table-num></table-num> Assembly Details
                        </th>
                    </tr>
                </thead>

                <tbody>
                    <tr>
                        <td>Contigs</td>
                        <td class="align-right">{{commas meta.contigs 0}}</td>
                    </tr>
                    <tr>
                        <td>GC Content</td>
                        <td class="align-right">{{default (toFixed meta.gc_content 2) 0}}</td>
                    </tr>
                    <tr>
                        <td>Plasmids</td>
                        <td class="align-right">{{commas meta.plasmids 0}}</td>
                    <tr>
                    </tr>
                        <td>Contig L50</td>
                        <td class="align-right">{{default meta.genome_metrics.L50 0}}</td>
                    </tr>
                    <tr>
                        <td>Genome Length</td>
                        <td class="align-right">{{commas meta.genome_length 0}} bp</td>
                    </tr>
                    <tr>
                        <td>Contig N50</td>
                        <td class="align-right">{{commas meta.genome_metrics.N50 0}}</td>
                    </tr>
                    <tr>
                        <td>Chromosomes</td>
                        <td class="align-right">{{commas meta.chromosomes 0}}</td>
                    </tr>

                    {{#if gto.job_data.assembly.job_id}}
                    <tr>
                        <td>Job ID</td>
                        <td class="align-right">{{default gto.job_data.assembly.job_id 'N/A'}}</td>
                    </tr>
                    <tr>
                        <td>Job Started</td>
                        <td class="align-right">{{datetime gto.job_data.assembly.start_time}}</td>
                    </tr>
                    <tr>
                        <td>Job Completed</td>
                        <td class="align-right">{{datetime gto.job_data.assembly.completion_time}}</td>
                    <tr>
                    </tr>
                        <td>Total Time</td>
                        <td class="align-right">{{elapsed gto.job_data.assembly.elapsed_time}}</td>
                    </tr>
                    <tr>
                        <td>Selected Recipe</td>
                        <td class="align-right">{{gto.job_data.assembly.parameters.recipe}}</td>
                    </tr>
                    {{/if}}

                </tbody>
            </table>

        </section>


        <!--
          ## Genome Annotation
        -->
        <section>
            <h2>Genome Annotation</h2>

            <p>
                The {{meta.genome_name}} genome was annotated
                using RAST tool kit (RASTtk)<ref>Brettin, et al. 2015</ref> and assigned a unique genome
                identifier of {{gto.id}}. This genome is in the superkingdom
                {{#eq gto.domain 'B'}}Bacteria{{/eq}}{{#eq gto.domain 'A'}}Archaea{{/eq}}
                and was annotated using genetic code {{gto.genetic_code}}.
                The taxonomy of this genome is:
            </p>

            <div style="width: 90%;" class="center">
                    {{#each gto.ncbi_lineage}} {{ this.[0] }} {{#unless @last }} > {{/unless }} {{/each}}
            </div>
            <br>

            {{#meta.feature_summary}}
            <p>
                This genome has
                    {{commas cds 0}} protein coding sequence{{plural cds}} (CDS),
                    {{commas tRNA 0}} transfer RNA (tRNA) gene{{plural tRNA}}, and
                    {{commas rRNA 0}} ribosomal RNA (rRNA) gene{{plural rRNA}}.
                The annotated features are summarized in <table-ref></table-ref>.
            </p>
            {{/meta.feature_summary}}

            <table class="med-table kv-table center">
                <thead class="table-header">
                    <tr>
                        <th colspan="2"><table-num></table-num> Annotated Genome Features</th>
                    </tr>
                </thead>

                <tbody>
                    {{#each annotationMeta}}
                        <tr>
                            <td>{{name}}</td>
                            <td class="align-right">
                                {{commas count 0}}
                            </td>
                        </tr>
                    {{/each}}

                    <tr>
                        <td>Job ID</td>
                        <td class="align-right">{{default gto.job_data.annotation.job_id 'N/A'}}</td>
                    </tr>
                    <tr>
                        <td>Job Started</td>
                        <td class="align-right">{{datetime gto.job_data.annotation.start_time}}</td>
                    </tr>
                    <tr>
                        <td>Job Completed</td>
                        <td class="align-right">{{datetime gto.job_data.annotation.completion_time}}</td>
                    <tr>
                    </tr>
                        <td>Total Time</td>
                        <td class="align-right">{{elapsed gto.job_data.annotation.elapsed_time}}</td>
                    </tr>
                </tbody>
            </table>


            <p>
                {{#pFeatures}}
                The annotation included
                    {{commas hypothetical 0}} hypothetical protein{{plural hypothetical}} and
                    {{commas function_assignment 0}} proteins with functional assignment{{plural function_assignment}}
                    (<table-ref></table-ref>).

                The proteins with functional assignments included
                    {{commas ec_assignment 0}} protein{{plural ec_assignment}} with Enzyme Commission (EC) numbers<ref>Schomburg, et al. 2004</ref>,
                    {{commas go_assignment 0}} with Gene Ontology (GO) assignments<ref>Ashburner, et al. 2000</ref>, and
                    {{commas pathway_assignment 0}} protein{{plural pathway_assignment}} that {{#if (eq pathway_assignment 1)}}was{{else}}were{{/if}} mapped to KEGG pathways<ref>Kanehisa, et al. 2016</ref>.
                PATRIC annotation includes two types of protein families<ref>Davis, et al. 2016</ref>,
                and this genome has
                    {{commas plfam_assignment 0}} protein{{plural plfam_assignment}} that belong to the genus-specific protein families (PLFams)
                for {{gto.ncbi_genus}}, and
                    {{commas pgfam_assignment 0}} protein{{plural pgfam_assignment}} that belong to the cross-genus protein families (PGFams).
                {{/pFeatures}}
            </p>


            {{#pFeatures}}
            <table class="med-table kv-table center">
                <thead class="table-header">
                    <tr>
                        <th colspan="2"><table-num></table-num> Protein Features</th>
                    </tr>
                </thead>

                <tbody>
                    <tr>
                        <td>Hypothetical proteins</td>
                        <td class="align-right">{{commas hypothetical 0}}</td>
                    </tr>
                    <tr>
                        <td>Proteins with functional assignments</td>
                        <td class="align-right">{{commas function_assignment 0}}</td>
                    </tr>
                    <tr>
                        <td>Proteins with EC number assignments</td>
                        <td class="align-right">{{commas ec_assignment 0}}</td>
                    <tr>
                    </tr>
                        <td>Proteins with GO assignments</td>
                        <td class="align-right">{{commas go_assignment 0}}</td>
                    </tr>
                    <tr>
                        <td>Proteins with Pathway assignments</td>
                        <td class="align-right">
                            {{commas pathway_assignment 0}}
                        </td>
                    </tr>
                    <tr>
                        <td>Proteins with PATRIC genus-specific family (PLfam) assignments</td>
                        <td class="align-right">
                            {{commas plfam_assignment 0}}
                        </td>
                    </tr>
                    <tr>
                        <td>Proteins with PATRIC cross-genus family (PGfam) assignments</td>
                        <td class="align-right">{{commas pgfam_assignment 0}}</td>
                    </tr>
                </tbody>
            </table>
            {{/pFeatures}}


            <p>
                A circular graphical display of the distribution of the genome annotations is provided
                (<fig-ref></fig-ref>).  This includes, from outer to inner rings, the contigs,
                CDS on the forward strand, CDS on the reverse strand, RNA genes,
                CDS with homology to known antimicrobial resistance genes, CDS with
                homology to know virulence factors, GC content and GC skew.
                The colors of the CDS on the forward and reverse strand indicate the subsystem
                that these genes belong to (see Subsystems below).
            </p>
	    {{#if truncatedCircos}}
	    <p> This genome has too many contigs to be rendered clearly. The circular display
	      has been limited to the {{ nContigsDrawn }} longest contigs of the {{ nContigs }} contigs in the genome. 
	      </p>
	    {{/if}}

            <fig-num></fig-num>
            <div class="circular-view-fig center" >{{{circularViewSVG}}}</div>

        </section>





        <!--
          ## Specialty Genes
        -->
        <section>
            <h2>Specialty Genes</h2>
            <p>
                Many of the genes annotated in {{gto.genome_id}} have homology to known
                transporters<ref>Saier Jr, et al. 2015</ref>, virulence factors<ref>Mao, et al. 2015; Chen, et al. 2016</ref>,
                drug targets<ref>Zhu, et al. 2009; Law, et al. 2014</ref>,
                and antibiotic resistance genes<ref>McArthur, et al. 2013</ref>.

                The number of genes and the specific source database where homology was found is provided (<table-ref></table-ref>).
            </p>

            <table class="sm-table kv-table center">
                <thead class="table-header">
                    <tr>
                        <th colspan="3"><table-num></table-num> Specialty Genes</th>
                    </tr>
                </thead>

                <tbody>
                    <tr class="row-header">
                        <td></td>
                        <td>Source</td>
                        <td class="align-right">Genes</td>
                    </tr>
                    {{#each specialtyGenes}}
                    <tr>
                        <td>{{type}}</td>
                        <td>{{source}}</td>
                        <td class="align-right">{{commas genes 0}}</td>
                    </tr>
                    {{/each}}
                </tbody>
            </table>
        </section>




        <!--
          ## AMR
        -->
        <section>
            {{#if amrPhenotypes}}
                <h2>Antimicrobial Resistance Analysis</h2>

                <p>
                    Antimicrobial resistance phenotypes refer to the resistance or susceptibility of a given
                    organism to one or more antibiotics. PATRIC provides AMR phenotype predictions
                    using custom built AdaBoost (adaptive boosting) machine learning
                    classifiers<ref>Davis, et al. 2016; Long, et al 2017</ref>.
                    The predicted AMR phenotypes fo this genome are summarized in <table-ref></table-ref>.
                </p>

                <table class="lg-table kv-table center">
                    <thead class="table-header">
                        <tr>
                            <th colspan="2"><table-num></table-num> Predicted Antimicrobial Resistance Phenotypes</th>
                        </tr>
                    </thead>

                    <tbody>
                        <tr class="row-header">
                            <td></td>
                            <td>Antibiotics</td>
                        </tr>
                        <tr>
                            <td>Resistant</td>
                            <td>{{join amrPhenotypes.resistant ', '}}</td>
                        </tr>
                        <tr>
                            <td>Susceptible</td>
                            <td>{{join amrPhenotypes.susceptible ', '}}</td>
                        </tr>
                    </tbody>

                </table>
            {{/if}}


            {{#if amrGenes}}
                <h2>Antimicrobial Resistance Genes</h2>

                <p>
                    The Genome Annotation Service in PATRIC uses k-mer-based AMR genes detection method, which
                    utilizes PATRIC’s curated collection of representative AMR gene sequence variants<ref>Wattam, et al. 2017</ref> and
                    assigns to each AMR gene functional annotation, broad mechanism of antibiotic resistance,
                    drug class and, in some cases, specific antibiotic it confers resistance to. Please note,
                    that the presence of AMR-related genes (even full length) in a given genome does not directly imply
                    antibiotic resistant phenotype.  It is important to consider specific AMR mechanisms and especially
                    the absence/presence of SNP mutations conveying resistance.  A summary of the AMR genes annotated in this
                    genome and corresponding AMR mechanism is provided in <table-ref></table-ref>.
                </p>


                <table class="lg-table kv-table center">
                    <thead class="table-header">
                        <tr>
                            <th colspan="2"><table-num></table-num> Antimicrobial Resistance Genes</th>
                        </tr>
                    </thead>

                    <tbody>
                        <tr class="row-header">
                            <td>AMR Mechanism</td>
                            <td>Genes</td>
                        </tr>
                        {{#amrGenes}}
                        <tr>
                            <td>{{label}}</td>
                            <td>{{join vals ', '}}</td>
                        </tr>
                        {{/amrGenes}}
                    </tbody>

                </table>
            {{/if}}

        </section>



        <!--
          ## Subsystem Analysis
        -->
        {{#if meta.subsystem_summary}}
        <section>
            <h2>Subsystem Analysis</h2>
            <p>
                A subsystem is a set of proteins that together implement a specific biological
                process or structural complex<ref>Overbeek, et al. 2005</ref> and PATRIC annotation
                includes an analysis of the subsystems unique to each genome.
                An overview of the subsystems for this genome is provided in <fig-ref></fig-ref>.
            </p>

            <fig-num></fig-num>
            <div class="center subsystem-fig" >{{{subsystemSVG}}}</div>
        </section>
        {{/if}}


        <!--
          ## Phylogenetic Analysis
        -->
        <section>
            <h2>Phylogenetic Analysis</h2>

            <p>
                The National Center for Biotechnology Information (NCBI) staff manually
                select and categorize reference and representative genomes, which they
                consider to be of high quality and importance to the research community.
                PATRIC provides the reference and representative genomes, and includes them in the
                phylogenetic analysis that is part of the Comprehensive Genome Analysis report.
                The closest reference and representative genomes to {{gto.genome_name}}
                were identified by Mash/MinHash<ref>Ondov, et al. 2016</ref>. PATRIC global
                protein families (PGFams)<ref>Davis, et al. 2016</ref> were selected from these
                genomes to determine the phylogenetic placement of this genome.  The
                protein sequences from these families were aligned with MUSCLE<ref>Edgar 2004</ref>,
                and the nucleotides for each of those sequences were mapped to the protein alignment.
                The joint set of amino acid and nucleotide alignments were concatenated into a data matrix,
                and RaxML<ref>Stamatakis 2014</ref> was used to analyze this matrix, with fast
                bootstrapping<ref>Stamatakis, et al. 2008</ref> was used to generate the support
                values in the tree (<fig-ref></fig-ref>).
            </p>


            <fig-num></fig-num>
            <div class="tree-fig center">{{{treeSVG}}}</div>


        </section>


        <!--
          ## Protein Families
        -->
        {{#if proteinFamily}} <!-- only shown if data is available -->
        <section>
            <h2>Unique Protein Families</h2>

            <p>
                In <table-rer></table-ref> we present protein families that the isolate has which
                all reference genomes lack.
            </p>


            <table class="lg-table center">
                <thead class="table-header">
                    <tr>
                        <th colspan="3"><table-num></table-num> Core Protein Families Missing in the Genome</th>
                    </tr>
                </thead>

                <tbody>
                    <tr class="row-header">
                        <td>Protein Family ID</td>
                        <td>Description</td>
                    </tr>
                    {{#proteinFamily}}
                    <tr>
                        <td>{{family_id}}</td>
                        <td>{{family_product}}</td>
                    </tr>
                    {{/proteinFamily}}
                </tbody>
            </table>
        </section>
        {{/if}}

        <h2>References</h2>
        <references></references>


    </body>
</html>
